basxconnect/core/layouts/admin_layout.py

```diff
@@ -57,4 +61,34 @@ def maintainance_package_layout(request):
 
 
 def maintenance_database_optimization(request):
-    return hg.H3("Coming Soon")
+    # get the database's size (in kilobytes)
+    # TODO: Add a button that runs the command
+    current_db_size = os.stat(os.getcwd() + "/db.sqlite3").st_size / 1000
+    previous_size = 0
+
+    optimize_btn = hg.FORM(
+        layout.form.CsrfToken(),
+        Button("Optimize", type="submit"),
+        hg.INPUT(type="hidden", name="previous-size", value=current_db_size),
+        method="POST",
+    )
```

Comment:
Okay, your solution works, but I would like to use the Django form-pattern (see https://docs.djangoproject.com/en/dev/topics/forms/). Instead of using ```hg.FORM``` I recommend you do something like this:

```python
class OptimizeForm(django.forms.Form):
    previous = django.forms.DecimalField(widget=django.forms.HiddenWidget, initial=current_db_size)

if request.method == "POST":
    form = OptimizeForm(request.POST)
    # existing code for VACUUM and calculation size
else:
    form = OptimizeForm()

return hg.BaseElement(bread.layout.forms.Form(form, bread.layout.forms.FormField("previous")))
```

This is just a conceptual example, you need to adjust it a litte bit to work.

Question: Why should we do it like this?
Answer: 1. Because we should rely on Djangos form mechanism for verification of form data. This is just a ground-rule because we do not have to worry about verifying user input (web application should always verify all data that is coming through a request). 2. Even though in this small example it does not make a big difference, I think it would be good to use the same pattern everywhere.